File: src/main/java/me/miki/shindo/management/event/impl/EventReceivePacket.java
package me.miki.shindo.management.event.impl;

import lombok.Getter;
import lombok.Setter;
import me.miki.shindo.management.event.Event;
import net.minecraft.client.network.NetHandlerPlayClient;
import net.minecraft.network.Packet;

@Getter
@Setter
public class EventReceivePacket extends Event {

    private Packet<?> packet;

    public EventReceivePacket(Packet<?> packet) {
        this.packet = packet;
    }

}

File: src/main/kotlin/me/miki/shindo/management/event/impl/EventReceivePacket.kt
package me.miki.shindo.management.event.impl

import me.miki.shindo.management.event.Event
import net.minecraft.network.Packet

class EventReceivePacket(
    private var packet: Packet<*>,
) : Event() {
    fun getPacket(): Packet<*> = packet

    fun setPacket(packet: Packet<*>) {
        this.packet = packet
    }
}
